soccer/gameplay/tactics/wall.py

```diff
@@ -0,0 +1,65 @@
+import composite_behavior
+import behavior
+import constants
+import robocup
+import main
+import enum
+import math
+import skills.move
+
+# This tactic builds a wall a certain distance from point A, blocking point B.
+class Wall(composite_behavior.CompositeBehavior):
+    class State(enum.Enum):
+        defense_wall = 1
+
+    def __init__(self,
+                 num_defenders = 3,                         # number of defenders we're making the wall with (default 3)
+                 curvature = 0,                             # curvature (in radians) of the wall 0 - pi/2
+                 mark_point = None,                         # what point we are defending against (default is ball)
+                 defender_point = robocup.Point(0, 0),      # what point we are defending (default is goal)
+                 defender_spacing = 3,                      # number of robot radii between the centers of the defenders in the wall
+                 dist_from_mark = 1):                       # distance from the mark point we want to build the wall
+        super().__init__(continuous=True)
+
+        self.number_of_defenders = num_defenders
+        self.curvature = curvature
+        self.mark_point = main.ball().pos if mark_point == None else mark_point
+        self.defense_point = defender_point
+        self.dist_from_mark = dist_from_mark
+        self.defender_spacing = defender_spacing
+
+        self.add_state(Wall.State.defense_wall, 
+                       behavior.Behavior.State.running)
+
+        self.add_transition(behavior.Behavior.State.start,
+                            Wall.State.defense_wall, lambda: True,
+                            "immediately")
+
+    def on_enter_defense_wall(self):
+        self.remove_all_subbehaviors()
+        midpoint = self.arc_midpoint()
+        angle = (math.atan2(midpoint.x - self.defense_point.x, midpoint.y - self.defense_point.y))
```

Comment:
You can subtract the Points then do a .angle() call.

eg `(b - a).angle()`

Reply:
Interestingly enough when I change my code to what you have written, the result breaks the wall's functionality. Is keep what I have ok, or should I use what you recommend for readability?

Reply:
Oh whoops, didn't realize you swapped the x and y in the atan2. Do a .rotate(pi/2).angle(). If that doesn't work I'll actually have to pull out pen and paper

Reply:
I'll look into it